docs/modules/ROOT/pages/tmail-backend/jmap-extensions/deletedMessagesVault.adoc

```diff
@@ -0,0 +1,185 @@
+= Deleted messages vault
+:navtitle: deleted messages vault
+
+Allows users to recover deleted messages in their vault.
+
+This is done by creating `EmailRecoveryAction` that restores deleted emails in the vault into users `Restored-messages`
+mailboxes.
+
+== Additions to the capability object
+
+Servers supporting the deleted message vaul extension need to advertise it through the session capabilities:
+....
+com:linagora:params:messages:vault
+....
+
+The associated object have the following fields:
+
+ - `maxRestoredEmails`: *Number | null*, defaulting to 5. Maximum number of emails a given `EmailRecoveryAction` would
+allow to recover.
+
+== EmailRecoveryAction
+
+This represents the recovery action for deleted items in the vault.
+
+It has the following properties:
+
+ - `id`: *Id* (server set) the object identifier.
+ - `deletedBefore`: *UTC date | null*. Immutable. If set this EmailRecoveryAction would restore only emails deleted before the aforementioned date.
+ - `deletedAfter`: *UTC date | null*. Immutable. If set this EmailRecoveryAction would restore only emails deleted after the aforementioned date.
+ - `receivedBefore`: *UTC date | null*. Immutable. If set this EmailRecoveryAction would restore only emails received before the aforementioned date.
+ - `receivedAfter`: *UTC date | null*. Immutable. If set this EmailRecoveryAction would restore only emails received after the aforementioned date.
+ - `hasAttachment`: *Boolean | null*. Immutable. If set this EmailRecoveryAction would restore only email whose `hasAttachment` property matches this field.
+ - `subject`: *String | null*. Immutable. If set this EmailRecoveryAction would restore only email whose `subject` property contains this field.
+ - `sender`: *String | null*. Immutable. If set this EmailRecoveryAction would restore only email whose `from` property matches this field. Needs to be a valid email address.
+ - `recipients`: *String[] | null*. Immutable. If set this EmailRecoveryAction would restore only email whose `to`, `cc` or `bcc` properties matches all of the mentionned email addresses.
+ Needs to be a valid email addresses.
+ - `status`: *Status*. The status of the task.
+
+*Status* can have the following values:
+
+ - `waiting`: the task is planned and waits to be executed
+ - `inProgress`: the task is currently being executed
+ - `done`: the task is done. It's execution did succeed.
+ - `failed`: the task execution failed.
+ - `canceled`: the task was canceled.
+
+=== EmailRecoveryAction/get
+
+Standard `/get` methods. the `ids` property cannot be null.
```

Comment:
`the `ids` property cannot be null.` 
-> Issue: how can the client track the submitted task between diff devices/ or even diff browsers?

Reply:
Answer: Not supported.